src/DotNetLightning.Core/Serialize/Msgs/Msgs.fs

```diff
@@ -1281,7 +1281,15 @@ type ErrorMessage =
                 ls.WriteWithLen(this.Data)
 
         member this.GetFailureMsgData() =
-            System.Text.ASCIIEncoding.ASCII.GetString(this.Data)
+            let isPrintableAscii =
+                not <| Array.exists (fun asciiByte -> asciiByte < 32uy) this.Data
+            if isPrintableAscii then
+                System.Text.ASCIIEncoding.ASCII.GetString(this.Data)
+            else
+                Array.fold
```

Comment:
nit: can you use Seq.fold instead? (equivalent of IEnumerable in the C# world) this way it will be future proof in case List is used instead of Array later on

Reply:
(same above about Array.exists -> Seq.exists)